Added tests for math function migrations

0x12CC opened the pull request:
This PR adds tests for the following math API functions:

- `__hadd2`
- `__habs`
- `__habs2`
- `labs`
- `llabs`

ziranzha commented on features/feature_case/math/math-direct.cu:
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
+
+  // CHECK: l = sycl::abs(l);
+  // CHECK-NEXT: ll = sycl::abs(ll);
+  l = labs(l);
```
It is better to check whether result is correct.

ziranzha commented on features/feature_case/math/math-direct.cu:
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
+  // CHECK: l = sycl::abs(l);
+  // CHECK-NEXT: ll = sycl::abs(ll);
+  l = labs(l);
+  ll = llabs(ll);
```
same as above.

ziranzha commented on features/feature_case/math/math-direct.cu:
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
+
+  // CHECK: h = sycl::fabs(h);
+  // CHECK-NEXT: h2 = sycl::fabs(h2);
+  h = __habs(h);
```
same as above.

ziranzha commented on features/feature_case/math/math-direct.cu:
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
+  // CHECK: h = sycl::fabs(h);
+  // CHECK-NEXT: h2 = sycl::fabs(h2);
+  h = __habs(h);
+  h2 = __habs2(h2);
```
same as above.

ziranzha commented on features/feature_case/math/math-emu.cu:
```diff
@@ -181,6 +183,8 @@ __global__ void math_emu() {
   // CHECK: sycl::rhadd(u, u2);
   __urhadd(u, u2);
+  // CHECK: h2 = h2 + h2
+  h2 = __hadd2(h2, h2);
```
same as above.

ziranzha reviewed (commented):
hcrp is tested in this PR or in another PR?

0x12CC commented:
I've updated the tests to assert the results and added a case for `hrcp`. The `__habs` and `__habs2` functions are in a separate file since they're only supported on GPU devices.

zhimingwang36 reviewed (approved):
LGTM

ShengchenJ reviewed (approved):
LGTM

ShengchenJ commented:
@0x12CC Please resolve the conflict. Thanks.

0x12CC committed:
Added tests for math function migrations

Signed-off-by: Michael Aziz <michael.aziz@intel.com>

0x12CC commented:
> @0x12CC Please resolve the conflict. Thanks.

This should be fixed now.
